src/app.js

```diff
@@ -0,0 +1,380 @@
+'use strict';
+
+const GRID_SIZE = 4;
+let gameGrid
+  = Array.from({ length: GRID_SIZE }, () => Array(GRID_SIZE).fill(0));
+let score = 0;
+let gameIsOver = false;
+
+const startButton = document.querySelector('.button.start');
+const winMessage = document.querySelector('.message-win');
+const loseMessage = document.querySelector('.message-lose');
+const startMessage = document.querySelector('.message-start');
+const restartButton = document.getElementById('restartButton');
+const scoreElement = document.querySelector('.game-score');
+
+function generateRandomNumber() {
+  const randomNumber = Math.random() < 0.1 ? 4 : 2;
+  const emptyCells = [];
+
+  for (let row = 0; row < GRID_SIZE; row++) {
+    for (let col = 0; col < GRID_SIZE; col++) {
+      if (gameGrid[row][col] === 0) {
+        emptyCells.push({
+          row,
+          col,
+        });
+      }
+    }
+  }
+
+  if (emptyCells.length > 0) {
+    const randomIndex = Math.floor(Math.random() * emptyCells.length);
+    const { row, col } = emptyCells[randomIndex];
+
+    gameGrid[row][col] = randomNumber;
+  }
+}
+
+function renderGame() {
+  const gameField = document.querySelector('.game-field');
+
+  for (let row = 0; row < GRID_SIZE; row++) {
+    for (let col = 0; col < GRID_SIZE; col++) {
+      const cellValue = gameGrid[row][col];
+      const cellElement = gameField.rows[row].cells[col];
+
+      cellElement.className = 'field-cell';
+
+      if (cellValue !== 0) {
+        cellElement.textContent = cellValue;
+        cellElement.classList.add(`field-cell--${cellValue}`);
+      } else {
+        cellElement.textContent = '';
+      }
+    }
+  }
+}
+
+function initGame() {
+  gameGrid = Array.from({ length: GRID_SIZE }, () => Array(GRID_SIZE).fill(0));
+
+  score = 0;
+  gameIsOver = false;
+
+  generateRandomNumber();
+  generateRandomNumber();
+
+  renderGame();
+
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+
+  startMessage.classList.remove('hidden');
+  startButton.classList.remove('hidden');
+  document.getElementById('restartButton').classList.add('hidden');
+}
+
+function restartGame() {
+  score = 0;
+  gameIsOver = false;
+
+  for (let row = 0; row < GRID_SIZE; row++) {
+    for (let col = 0; col < GRID_SIZE; col++) {
+      gameGrid[row][col] = 0;
+    }
+  }
+
+  generateRandomNumber();
+  generateRandomNumber();
+
+  renderGame();
+
+  winMessage.classList.add('hidden');
+  loseMessage.classList.add('hidden');
+
+  startMessage.classList.add('hidden');
+  startButton.classList.add('hidden');
+
+  document.getElementById('restartButton').classList.remove('hidden');
+  document.querySelector('.game-score').textContent = score;
+
+  restartButton.blur();
+}
+
+restartButton.addEventListener('click', function() {
```

Comment:
You should use an arrow function for the event listener to make your code more concise and readable.